.github/copilot-instructions.md

```diff
@@ -0,0 +1,282 @@
+│   ├── actionAngle*.py   # Various action-angle classes
+│   └── actionAngle*_c_ext/  # C extensions for performance
+├── df/                   # Distribution functions
+├── orbit/                # Orbit integration and analysis
+│   ├── Orbit.py          # Main Orbit class
+│   └── orbit_c_ext/      # C extensions for orbit integration
+├── potential/            # Gravitational potentials
+│   ├── *Potential.py     # Individual potential classes
+│   ├── mwpotentials.py   # Milky Way models
+│   └── potential_c_ext/  # C extensions for potentials
+├── snapshot/             # N-body snapshot support
+└── util/                 # Utility functions
+
+tests/                    # Test suite (~87,828 lines)
+├── test_*.py             # Test modules (match galpy subpackages)
+└── conftest.py           # pytest configuration
+
+doc/                      # Sphinx documentation
+├── Makefile              # Builds to ../../galpy-docs/
+└── source/               # Documentation source files
+```
+
+### Key Configuration Files
+- `.coveragerc`: Coverage omit patterns (snapshot, amuse, some util files)
+- `.readthedocs.yaml`: Documentation build (Python 3.11, Ubuntu 22.04)
+- `.github/workflows/build.yml`: Linux/macOS CI (most comprehensive)
+- `.github/workflows/build_windows.yml`: Windows CI
+- `.github/workflows/wheels.yml`: Wheel building for releases
+
+## CI/CD and Validation
+
+### GitHub Actions Workflows
+
+**Main Build Workflow** (`.github/workflows/build.yml`):
+- Runs on: Ubuntu, macOS (Intel)
+- Python versions: 3.9, 3.10, 3.11, 3.12, 3.13
+- **Test splitting**: Tests are split into ~15 jobs per Python version for parallel execution
+- **OpenMP disabled on Linux**: CI uses `GALPY_COMPILE_NO_OPENMP=1`
+- **Coverage enabled on Linux**: Uses `GALPY_COMPILE_COVERAGE=1` with lcov
+- **Typical test warnings**: DeprecationWarning and FutureWarning treated as errors
+- **astropy warnings**: Also treated as errors when astropy is installed
+
+**Windows Build Workflow** (`.github/workflows/build_windows.yml`):
+- Uses micromamba for GSL installation via conda-forge
+- Sets GSL environment variables: `INCLUDE`, `LIB`, `LIBPATH`
+- Simpler test splits (no pynbody support)
+
+**Pre-commit CI**:
+- Runs automatically on PRs
+- Auto-fixes issues when possible
+
+### Coverage Requirements
+
+- Target coverage: Very high (see badges in README)
+- Coverage reports uploaded to codecov.io
+- C code coverage tracked via lcov on Linux
+
+## Common Workflows
+
+### Adding a New Potential
+
+1. Create new potential class in `galpy/potential/NewPotential.py`
+2. Follow existing potential structure (inherit from appropriate base class)
+3. Implement required methods: `_evaluate`, `_Rforce`, `_zforce`, etc.
+4. Add tests to `tests/test_potential.py`
+5. Add API documentation page in `doc/source/reference/potentialnew.rst` (2-line file)
+6. **Optional**: Add C implementation following `README.dev` steps
+7. Run tests: `pytest tests/test_potential.py -v -k NewPotential`
```

Comment:
Also add running the relevant orbit tests `pytest tests/test_orbit.py -v -k NewPotential`